packages/flutter/lib/src/cupertino/date_picker.dart

```diff
@@ -1053,6 +1139,29 @@ class _CupertinoTimerPickerState extends State<CupertinoTimerPicker> {
     );
   }
 
+  /// Sets the picker to be the initial time.
+  void _desiredDurationChanged() {
```

Comment:
This logic looks very similar to what is in `init()` - should this method be invoked from there, too?

Reply:
Indeed. Put the initialTimerDuration value into a newly created controller if the controller value is null. Then just invoke this.